RallyHereDebugTool/Source/Private/RHDTW_PlayerSettings.cpp

```diff
@@ -144,6 +150,69 @@ void FRHDTW_PlayerSettings::DoModifySettings()
 	}
 }
 
+void FRHDTW_PlayerSettings::DoSettingsTypes()
+{
+	ULocalPlayer* pLocalPlayer = GetFirstSelectedLocalPlayer();
+	if (pLocalPlayer == nullptr)
+	{
+		ImGui::Text("Requesting Types requires a logged in local player for Auth Context");
+		return;
+	}
+
+	URH_LocalPlayerSubsystem* pRH_LocalPlayerSubsystem = pLocalPlayer->GetSubsystem<URH_LocalPlayerSubsystem>();
+	if (pRH_LocalPlayerSubsystem == nullptr)
+	{
+		ImGui::Text("Requesting Types requires a logged in local player for Auth Context");
+		return;
+	}
+
+	if (!pRH_LocalPlayerSubsystem->GetAuthContext()->IsLoggedIn())
+	{
+		ImGui::Text("Requesting Types requires a logged in local player for Auth Context");
+		return;
+	}
+
+	if (ImGui::Button("Get Setting Types"))
+	{
+		typedef RallyHereAPI::Traits_GetConfigForAllSettingTypes TGetConfigSettings;		
```

Comment:
I just put the call in here and cached the data on the debug tool itself, I thought about putting it somewhere in the Integration Layer but was not sure where was a good spot, as Settings is kinda a subpiece of PlayerInfos which there are many of, so felt a bit weird on them, up for suggestions

Reply:
Are there Catalog Templates? What uses them?

Session Type Templates are cached in various ways on the URH_MatchmakingBrowserCache I believe.  

The problem is Settings in general isn't its own Class, so it doesn't really have a nice home for code.  Which may be the solution or we just leave this call where I put it

Reply:
There are no catalog templates, but all the components of catalog (loot, item, vendor, etc) are pieces of configuration from the dev portal that clients like to have a view of.  Session templates and Setting Types are a similar kind of data.  Stuff that the game teams configure in the Dev Portal/API, and may want their clients to pull a copy of that down to do client side validation against.

I'm in favor of giving it its own class.  Even if we only use it to cache this configuration data.